### PR TITLE
Add example for Ray Serve deployment plugin

### PR DESCRIPTION
**Pull request by architkulkarni**
## What changes are proposed in this pull request?

This PR adds a basic example of using the MLflow Ray Serve deployment plugin.  In the example, a model is trained to classify the Iris dataset using `sklearn`. Next, the model is deployed on Ray Serve and scaled up.

This PR also adds the Ray Serve plugin to the list of known third-party deployment plugins in the MLflow documentation.

## How is this patch tested?

A test has been added alongside the rest of the example tests.
### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)
Add usage example for MLflow Ray Serve deployment plugin
### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [x] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by dmatrix** [comment]
@architkulkarni You will want to [sign your commits](https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#sign-your-work), for the DCO test to succeed.

**Comment by architkulkarni** [comment]
Thanks @dmatrix, apologies for the oversight.  It's fixed now.

**Review by dmatrix — CHANGES_REQUESTED** [review]
Shouldn't that be

**Review by dmatrix — CHANGES_REQUESTED** [review]
Shouldn't this be the `-m model-uri?`

File: examples/ray_serve/README.md
Comment (architkulkarni) [reply examples/ray_serve/README.md:13]: Thanks, that makes sense!  It should be more readable now.

**Review by dmatrix — COMMENTED** [review]
LGTM with the sequence of steps.

File: examples/ray_serve/train_model.py
Comment (harupy) [inline examples/ray_serve/train_model.py:25]: Should we avoid shuffling `data` and `target` separately?

Comment (harupy) [reply examples/ray_serve/train_model.py:25]: I think we can use https://scikit-learn.org/stable/modules/generated/sklearn.utils.shuffle.html

File: examples/ray_serve/README.md
Comment (dmatrix) [inline examples/ray_serve/README.md:13]: should that be relative path `mlflow/examples/ray_serv/`

File: examples/ray_serve/README.md
Comment (dmatrix) [inline examples/ray_serve/README.md:29]: Should that pronoun me `your`? Our suggests it's maintained outside, elsewhere.

File: examples/ray_serve/README.md
Comment (dmatrix) [inline examples/ray_serve/README.md:35]: Perhaps `can` is better than using tentative `may`

File: examples/ray_serve/README.md
Comment (dmatrix) [inline examples/ray_serve/README.md:39]: maybe say:  which consists of the registered model name and version in the Model Registry.

File: examples/ray_serve/train_model.py
Comment (dmatrix) [inline examples/ray_serve/train_model.py:15]: I know I wrote this code bit. Perhaps call this function `eval_and_print_model_info(...)`

**Review by dmatrix — COMMENTED** [review]
Added minor nits and suggestions.

**Comment by architkulkarni** [comment]
Thanks a lot for the detailed review!  I implemented the suggestions and got rid of `print_model_info` since it isn't used in the example and I forgot to remove it earlier.

**Review by dmatrix — APPROVED** [review]
LGTM :-) after the test failures have been resolved.

@harupy If all looks good to you, we can merge it. Not sure if it's too late for the 1.14 release. But will leave that up to you.

**Review by harupy — APPROVED** [review]
LGTM :)

**Comment by harupy** [comment]
@architkulkarni The flavors check should be fixed by merging the master branch.

**Comment by architkulkarni** [comment]
@harupy Thanks, it worked!

**Pull request merged**
